secure-connection/register.go

```diff
@@ -17,15 +17,27 @@ func register(err error, client *nex.Client, callID uint32, stationUrls []*nex.S
 	client.SetConnectionID(nextConnectionID)
 
 	localStation := stationUrls[0]
-	publicStation := localStation.Copy()
 
-	publicStation.SetAddress(client.Address().IP.String())
-	publicStation.SetPort(uint32(client.Address().Port))
-	publicStation.SetNatf(0)
-	publicStation.SetNatm(0)
-	publicStation.SetType(3)
+	// * A NEX client can set the public station URL by setting two URLs on the array
+	var publicStation *nex.StationURL
+	if len(stationUrls) > 1 {
+		publicStation = stationUrls[1]
+	} else {
+		publicStation = localStation.Copy()
+
+		publicStation.SetAddress(client.Address().IP.String())
+		publicStation.SetPort(uint32(client.Address().Port))
+		publicStation.SetNatf(0)
+		publicStation.SetNatm(0)
+		publicStation.SetType(3)
+	}
+
+	localStation.SetPID(client.PID())
 	publicStation.SetPID(client.PID())
 
+	localStation.SetRVCID(client.ConnectionID())
+	publicStation.SetRVCID(client.ConnectionID())
+
```

Comment:
Has this been tested on a wide number of games? I tried always setting these values (PID and connection ID) on all the stations in `GetSessionURLs` while working on Minecraft, and it seemed to break PIA in Mario Kart 8. It could have been a fluke, but I tried a few times and the guest couldn't connect to the host until I reverted those changes

Reply:
> I tried always setting these values (PID and connection ID) on all the stations in GetSessionURLs while working on Minecraft, and it seemed to break PIA in Mario Kart 8.

That's weird. AFAIK, the PID and connection ID should be set on the stations anyway when calling `ReportNATProperties`. We can't rely on that for NEX 2.0 games though, because they don't seem to call that method at any time, so I'm setting them on `Register` aswell.

Reply:
I'll merge this for now, and check Mario Kart 8 again. If it breaks, then we have to find a different solution